src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -213,6 +214,29 @@ export default class UniversalDiscoveryModule extends Component {
         document.body.dispatchEvent(event);
     }
 
+    /**
+     * Sets loading status of bookmark
+     *
+     * @method setBookmarkLoadingStatus
+     * @param {String} locationId
+     * @param {Boolean} isLoading
+     * @memberof UniversalDiscoveryModule
+     */
+    setBookmarkLoadingStatus(locationId, isLoading) {
+        this.loadingBookmarksLocationsIds[locationId] = isLoading;
+    }
+
+    /**
+     * Returns true if bookmark is loading, otherwise returns false
+     *
+     * @method isBookmarkLoading
+     * @param {String} locationId
+     * @memberof UniversalDiscoveryModule
+     */
+    isBookmarkLoading(locationId) {
+        return locationId in this.loadingBookmarksLocationsIds && this.loadingBookmarksLocationsIds[locationId];
```

Comment:
I think `!!this.loadingBookmarksLocationsIds[locationId]` is enough.

Reply:
Good point